src/Corax/Pipeline/Parsing/VectorParsers.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using Sparrow;
+
+
+namespace Corax.Pipeline.Parsing
+{
+    internal static class VectorParsers
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int FindFirstNonAscii(ReadOnlySpan<byte> buffer)
+        {
+            ref byte bufferStart = ref MemoryMarshal.GetReference(buffer);
+
+            nint pos = 0;
+            nint len = buffer.Length;
+
+            if (AdvInstructionSet.IsAcceleratedVector512)
+            {
+                // process in blocks of 64 bytes when possible
+                const int N512 = 64;
+                
+                for (; pos + N512 < len; pos += N512)
+                {
+                    Vector512<byte> v1 = Vector512.LoadUnsafe(ref Unsafe.AddByteOffset(ref bufferStart, pos));
+                    Vector512<byte> v2 = Vector512.LoadUnsafe(ref Unsafe.AddByteOffset(ref bufferStart, pos + N512));
+                    var v = Vector512.BitwiseOr(v1, v2);
+
+                    Vector512<byte> mask = Vector512.Create((byte)0x80);
+                    Vector512<byte> result = Vector512.BitwiseAnd(v, mask);
+
+                    if (result != Vector512<byte>.Zero)
+                        break;
```

Comment:
I wonder if it's a better idea to go directly into scalar code, instead of `Vec512 -> Vec256 -> Vec128 -> Scalar`.